g2p_programs/models/managers/payment_manager.py

```diff
@@ -81,24 +83,37 @@ def prepare_payments(self, cycle):
         )
         # _logger.info("DEBUG! payments_to_create: %s", payments_to_create)
 
-        # Todo: Create payment batch
-        # batch = self.env["g2p.paymentbatch"].create()
-
         ctr = 0
+        vals = []
+        payments_to_add_ids = []
         for entitlement_id in entitlements_with_payments_to_create:
-            self.env["g2p.payment"].create(
+            payment = self.env["g2p.payment"].create(
                 {
                     "entitlement_id": entitlement_id.id,
                     "cycle_id": entitlement_id.cycle_id.id,
                     "amount_issued": entitlement_id.initial_amount,
                     "payment_fee": entitlement_id.transfer_fee,
                     "state": "issued",
                     # "account_number": self._get_account_number(entitlement_id),
-                    # "batch_id": batch.id,
                 }
             )
+            vals.append((4, payment.id))
```

Comment:
What does `4` mean in this context?

Reply:
@gonzalesedwin1123 maybe replace it by `Commands.UNLINK` https://github.com/odoo/odoo/blob/15.0/odoo/fields.py#L3013 those numbers are also confusing for me.

Reply:
It is based on this Odoo ORM reference when populating one2many and many2many data directly in the parent model:
`(0, 0,  { values })    link to a new record that needs to be created with the given values dictionary
(1, ID, { values })    update the linked record with id = ID (write *values* on it)
(2, ID)                remove and delete the linked record with id = ID (calls unlink on ID, that will delete the object completely, and the link to it as well)
(3, ID)                cut the link to the linked record with id = ID (delete the relationship between the two objects but does not delete the target object itself)
(4, ID)                link to existing record with id = ID (adds a relationship)
(5)                    unlink all (like using (3,ID) for all linked records)
(6, 0, [IDs])          replace the list of linked IDs (like using (5) then (4,ID) for each ID in the list of IDs)`

Reply:
I will try the Commands

Reply:
Both of these 
`            vals.append((Command.LINK, payment.id))
            vals.append(Command.link(payment.id))
`
works by importing Command from odoo. I used the Command.link function instead of the constant Command.LINK

Reply:
yes, better